test/unit/BCoWFactory.t.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {Base, BaseBFactory_Unit_Constructor, BaseBFactory_Unit_NewBPool} from './BFactory.t.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {IBCoWFactory} from 'interfaces/IBCoWFactory.sol';
+import {IBCoWPool} from 'interfaces/IBCoWPool.sol';
+import {IBFactory} from 'interfaces/IBFactory.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
+import {MockBCoWFactory} from 'test/manual-smock/MockBCoWFactory.sol';
+
+abstract contract BCoWFactoryTest is Base {
+  address public solutionSettler = makeAddr('solutionSettler');
+  bytes32 public appData = bytes32('appData');
+
+  function _configureBFactory() internal override returns (IBFactory) {
+    vm.mockCall(solutionSettler, abi.encodePacked(ISettlement.domainSeparator.selector), abi.encode(bytes32(0)));
+    vm.mockCall(
+      solutionSettler, abi.encodePacked(ISettlement.vaultRelayer.selector), abi.encode(makeAddr('vault relayer'))
+    );
+    vm.prank(owner);
+    return new MockBCoWFactory(solutionSettler, appData);
+  }
+
+  function _bPoolBytecode() internal virtual override returns (bytes memory _bytecode) {
+    vm.skip(true);
```

Comment:
I'm not getting any test skips with the `Deploy_New_BPool`

    > FOUNDRY_FUZZ_RUNS=10 nice forge test --mt Deploy_NewBPool |grep SKIP
    
we are not extending the `BFactory_Internal_NewBPool` contract

also, we could save the fuzzed constructor inputs into storage, deploy the contract here reading them, and then return `address(pool).code`